routing/routing_helpers.cpp

```diff
@@ -18,6 +20,75 @@ namespace routing
 using namespace std;
 using namespace traffic;
 
+void FillSegmentInfo(vector<Segment> const & segments, vector<Junction> const & junctions,
+                     Route::TTurns const & turnDirs, Route::TStreets const & streets,
+                     Route::TTimes const & times, shared_ptr<TrafficStash> const & trafficStash,
+                     vector<Route::SegmentInfo> & segmentInfo)
+{
+  CHECK_EQUAL(segments.size() + 1, junctions.size(), ());
+  CHECK(!turnDirs.empty(), ());
+  CHECK(!times.empty(), ());
+
+  CHECK(std::is_sorted(times.cbegin(), times.cend(), my::LessBy(&Route::TTimeItem::first)), ());
+  CHECK(std::is_sorted(turnDirs.cbegin(), turnDirs.cend(), my::LessBy(&turns::TurnItem::m_index)), ());
+  CHECK(std::is_sorted(streets.cbegin(), streets.cend(), my::LessBy(&Route::TStreetItem::first)), ());
+
+  segmentInfo.clear();
+  if (segments.empty())
+    return;
+
+  segmentInfo.reserve(segments.size());
+  // Note. |turnDirs|, |streets| and |times| have point()|junctions| based index.
+  // On the other hand turns, street names and times are considered for further end of |segments| after conversion
+  // to |segmentInfo|. It means that street, turn and time information of zero point will be lost after
+  // conversion to |segmentInfo|.
+  size_t turnIdx = turnDirs[0].m_index != 0 ? 0 : 1;
```

Comment:
Yes. As it's written at the comment above turn information at 0 point is lost. So if there's any turn at zero point (that mean if turnDirs[0].m_index == 0) the turn will not passed to segment info vector.

The loop below goes through all the segments. We are interested in information at the further end of the segment. So if 0 segment is considered the information at point 1 is considered.

At the loop below there's a condition:
```
if (... turns[turnIdx].m_index == i + 1)
{
  curTurn = turns[turnIdx];
  ++turnIdx;
}
```
For i = 0 (at the time turnIdx == 0 as well) we need to compare turns[0/*turnIdx*/].m_index with 1. Image the rare situation when turns[0].m_index == 0. If |turnIdx| is set to 0 |turnIdx| will be zero during all the loop.

So we need the like to skip zero turn.